site/content/docs/07-state-management.md

```diff
@@ -37,6 +40,10 @@ express() // or Polka, or a similar framework
 	.listen(process.env.PORT);
 ```
 
-The `session` option is a function which returns data, or a `Promise` of data, which must be serializable (using [devalue](https://github.com/Rich-Harris/devalue)), meaning that it must not contain functions or custom classes, just built-in JavaScript data types.
+`req` is an `http.ClientRequest` and `res` an `http.ServerResponse`.
+
+The session data must be serializable. This means it must not contain functions or custom classes, just built-in JavaScript data types. [devalue](https://github.com/Rich-Harris/devalue) is used for serialization.
+
+The `session` function may return a `Promise` (or, equivalently, be `async`).
 
-> Note that if you return a `Promise` from this function (or use an `async` function), it will be re-awaited for **every** server-rendered page route.
\ No newline at end of file
+> If `session` returns a `Promise`, every server-rendered page route will wait for it to resolve. This can slow down your application significantly.
```

Comment:
I prefer the old text here for a few reasons. It starts with "Note that" to show it's something to pay attention to, informed users that `async` functions were also affected by this, it bolded every to call attention to it, it didn't try to predict the impact to the user's application

It was a little verbose though and I think could be made slightly more concise:

```suggestion
> Note that if `session` returns a `Promise` (or is `async`), it will be re-awaited for on **every** server-rendered page route.
```

Reply:
Sure. I just found the bolding shouty. I'm honestly not sure what it is really warning of. I assumed it was talking about the potential performance impact. It would be helpful to spell out why the await is or can be a problem.